### PR TITLE
Two singularities in the world can link on some occasions

### PR DESCRIPTION
**Pull request by ghost**
[content]
When they're linked, the outwards facing one becomes a "white hole", that repels items from it rather than attracting and eating them, and the old inward one teleports them to the outward one on consumption.

:cl:
 * rscadd: Two singularities existing at once in the world can now "link" with each other on creation (1% chance) or on reaction with bluespace crystals (10% chance if not flawless, artifical ones won't work) as a one directional wormhole, with the outwards facing one repelling items instead of attracting, and teleporting items to it from the inward one instead of them being consumed by that. They may also spew time, back into the universe.

### CODE REVIEW

**Comment by Eneocho** [comment]
Make it so there's a chance for items to get transported between the linked singulos (instead of eaten) and we have a deal.

**Comment by ghost** [comment]
> Make it so there's a chance for items to get transported between the linked singulos (instead of eaten) and we have a deal.

I made that a thing too in the PR

**Comment by boy2mantwicethefam** [comment]
I dunno, maybe if you made one of the linked singularities eject items absorbed by the other sucking singularity instead of having them obliterated then that'd be cool

**Comment by Eneocho** [comment]
If the second singulo spits/repels stuff, how is it supposed to grow? Does it feed on the first singulo's power?

**Comment by Kurfursten** [comment]
I do get the joke about "Soutularity" but it seems too silly to have as an actual name.

Why not call it a "White Hole"? After all, it is the theoretical concept you're borrowing from

**Comment by ghost** [comment]
> I do get the joke about "Soutularity" but it seems too silly to have as an actual name.
> 
> Why not call it a "White Hole"? After all, it is the theoretical concept you're borrowing from

So what is it?

**Comment by Kurfursten** [comment]
> > I do get the joke about "Soutularity" but it seems too silly to have as an actual name.
> > Why not call it a "White Hole"? After all, it is the theoretical concept you're borrowing from
> 
> So what is it?

I've never seen it before, no one has.

**Comment by gurfan** [comment]
this could have been a cool idea but I can't imagine a lazier implementation

**Comment by ghost** [comment]
> this could have been a cool idea but I can't imagine a lazier implementation

What's lazy about it?

**Comment by Hinaichigo** [comment]
Maybe there'd be something more fun than having it as a random chance. Like a player would have to do something specific to and somewhat complicated to generate a white hole.

**Comment by Eneocho** [comment]
You could make a black hole by something something white paint. You throw white paint onto the black hole and it goes white.

**Comment by Hinaichigo** [comment]
The concept of a white hole issuing forth speech is very interesting.

**Comment by Kurfursten** [comment]
You didn't change the name though!

**Comment by Blithering** [comment]
So what is it?

**Comment by ghost** [comment]
> You didn't change the name though!

Did just now.

File: code/modules/power/singularity/singularity.dm [hunk 1]
```diff
@@ -52,6 +56,39 @@ var/list/global_singularity_pool
 	if(!global_singularity_pool)
 		global_singularity_pool = list()
 	global_singularity_pool += src
+	if(prob(10) && global_singularity_pool.len > 1)
+		var/obj/machinery/singularity/other = pick(global_singularity_pool)
+		if(other && other != src)
```
Comment (Kurfursten) [inline code/modules/power/singularity/singularity.dm]: `&& !other.repels`

Comment (ghost) [reply code/modules/power/singularity/singularity.dm]: Done

**Comment by ghost** [comment]
> Maybe there'd be something more fun than having it as a random chance. Like a player would have to do something specific to and somewhat complicated to generate a white hole.

Changed it to that, with bluespace crystals.

**Comment by Hinaichigo** [comment]
Honestly it might be cool to keep the random chance as well as having it work with the crystals.

**Comment by Eneocho** [comment]
Any level of bluespace crystal?

**Comment by Hinaichigo** [comment]
Also it might be cool if the animation of the soutgulo is reversed so it looks like it's spiraling outwards or something.

**Comment by Eneocho** [comment]
I'd really change the name, soutgularity just doesn't feel quite right.

**Comment by Eneocho** [comment]
Could you make it so there's a small chance for a singulo to turn white hole if it eats a can of white paint?

**Comment by pedr69** [comment]
where do you get flawless bluespace anyway? apart from spacemart and one of the vaults

**Comment by Eneocho** [comment]
The old vendor event can have one too.

**Comment by Kurfursten** [comment]
Given the rarity of a flawless crystal (and its other uses which are EXTREMELY powerful), it's fine (in my view) that this be a 100% chance.

That said, it's pretty easy to farm natural crystals... but not trivial. Probably fine.

**Pull request merged**